FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -18,12 +18,14 @@ namespace FaunaDB.Client
     class DefaultClientIO : IClientIO
     {
         readonly HttpClient client;
-        readonly string authHeader;
+        readonly AuthenticationHeaderValue authHeader;
+        readonly bool useRecursiveRefs;
 
-        internal DefaultClientIO(HttpClient client, string secret)
+        internal DefaultClientIO(HttpClient client, AuthenticationHeaderValue authHeader, bool useRecursiveRefs)
```

Comment:
Using a flag like `useRecursiveRefs` to enabled api 2.1 can be a little bit dangerous as that api version could enable more than just the recursive refs as we extend its capabilities while still supporting the version 2.0.

It will be more clear if we treat this as a api version parameter, possibly a enum.